Microsoft.HealthVault.Fhir/Transformers/MedicationRequestToHealthVault.cs

```diff
@@ -0,0 +1,193 @@
+            {
+                throw new NotSupportedException($"{nameof(MedicationRequest)} needs to have an embedded Requester Agent");
+            }
+
+            return hvMedication;
+        }
+
+        private static int? GetDaysFromDuration(Hl7.Fhir.Model.Duration duration)
+        {
+            if (duration == null)
+            {
+                return null;
+            }
+            if (duration.Value.HasValue)
+            {
+                switch (duration.Code)
+                {
+                    case "day":
+                        return decimal.ToInt32(duration.Value.Value);
+                    case "week":
+                    case "month":
+                    case "year":
+                        double value = decimal.ToDouble(duration.Value.Value);
+                        var fromUnit = UnitsNet.Duration.ParseUnit(duration.Code);
+                        var unitsNetDuration = UnitsNet.Duration.From(value, fromUnit);
+                        return Convert.ToInt32(unitsNetDuration.Days);
+                }
+            }
+            throw new NotImplementedException();
+        }
+
+        private static CodableValue GetSubstitutionCode(MedicationRequest medicationRequest, Prescription prescription)
+        {
+            if (medicationRequest.Substitution.Allowed.HasValue)
+            {
+                switch (medicationRequest.Substitution.Allowed.Value)
+                {
+                    case true:
+                        return HealthVaultMedicationSubstitutionCodes.SubstitutionPermitted;
+                    case false:
+                        return HealthVaultMedicationSubstitutionCodes.DispenseAsWritten;
+                }
+            }
+            throw new NotImplementedException();
+        }
+
+        private static FhirMedication ExtractEmbeddedMedication(MedicationRequest medicationRequest)
+        {
+            switch (medicationRequest.Medication)
+            {
+                case ResourceReference medicationReference:
+                    if (!medicationReference.IsContainedReference)
+                    {
+                        throw new NotImplementedException();
+                    }
+                    var fhirMedication = medicationRequest.Contained.First(domainResource
+                             => medicationReference.Matches(domainResource.GetContainerReference())) as FhirMedication;
+                    return fhirMedication ?? new FhirMedication();
+                case CodeableConcept medicationCodeableConcept:
+                    return new FhirMedication
+                    {
+                        Code = medicationCodeableConcept
+                    };
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
+
+        private static Practitioner ExtractEmbeddedPractitioner(MedicationRequest medicationRequest)
+        {
+            var agentReference = medicationRequest.Requester?.Agent;
```

Comment:
>agentReference [](start = 16, length = 14)

Any chance Agent is null? We should check for null

Reply:
Fixed